### PR TITLE
#526107 | Define example Rendering Variants for existing Headless components

### PR DESCRIPTION
**Pull request by alan-null**

### CODE REVIEW

File: packages/create-sitecore-jss/src/templates/nextjs-sxa/src/components/Image.tsx [hunk 1]
```diff
@@ -20,45 +21,34 @@ type ImageProps = {
...
+          }
+          <Text tag="span" className="image-caption field-imagecaption" field={props.fields.ImageCaption} />
         </div>
       </div>
     );
   }
 
   return <ImageDefault {...props} />;
-};
+};
```
Comment (illiakovalenko) [inline packages/create-sitecore-jss/src/templates/nextjs-sxa/src/components/Image.tsx:54]: Add EOL

File: packages/create-sitecore-jss/src/templates/nextjs-sxa/src/components/Title.tsx [hunk 2]
```diff
@@ -72,10 +80,16 @@ export const Default = (props: TitleProps): JSX.Element => {
...
 
   return (
-    <ComponentContent styles={props.params.styles}>
-      <Link field={link}>
-        <Text field={text} />
-      </Link>
+    <ComponentContent styles={props.params.styles.trimEnd()}>
+      <>
+        {(sitecoreContext.pageState === 'edit') ?
+          <Text field={text} />
```
Comment (illiakovalenko) [inline packages/create-sitecore-jss/src/templates/nextjs-sxa/src/components/Title.tsx:86]: Did you try to run `lint` command? is it correct formatting?

File: packages/create-sitecore-jss/src/templates/nextjs-sxa/src/components/Title.tsx [hunk 3]
```diff
@@ -72,10 +80,16 @@ export const Default = (props: TitleProps): JSX.Element => {
   }
 
   return (
-    <ComponentContent styles={props.params.styles}>
-      <Link field={link}>
-        <Text field={text} />
-      </Link>
+    <ComponentContent styles={props.params.styles.trimEnd()}>
```
Comment (illiakovalenko) [inline packages/create-sitecore-jss/src/templates/nextjs-sxa/src/components/Title.tsx:83]: In general, I'm wondering - is it possible to trim these strings on the server? For me it looks strange that you need to trim values everywhere, and it means that customer will need to do the same everywhere

File: packages/create-sitecore-jss/src/templates/nextjs-sxa/src/components/Navigation.tsx [hunk 4]
```diff
@@ -64,26 +66,32 @@ const Navigation = (props: NavigationProps): JSX.Element => {
 };
 
 const NavigationList = (props: NavigationProps) => {
+  var title;
```
[the remark this answers is not in the document]
Comment (alan-null) [reply packages/create-sitecore-jss/src/templates/nextjs-sxa/src/components/Navigation.tsx]: Done

File: packages/create-sitecore-jss/src/templates/nextjs-sxa/src/components/Navigation.tsx [hunk 5]
```diff
@@ -64,26 +66,32 @@ const Navigation = (props: NavigationProps): JSX.Element => {
...
+  if (props.fields.NavigationTitle) {
+    title = props.fields.NavigationTitle.value?.toString();
+  } else if (props.fields.Title) {
+    title = props.fields.Title.value?.toString();
+  } else {
+    title = props.fields.DisplayName;
+  }
+
+  var link = <>
```
[the remark this answers is not in the document]
Comment (alan-null) [reply packages/create-sitecore-jss/src/templates/nextjs-sxa/src/components/Navigation.tsx]: Done

**Review by illiakovalenko — APPROVED** [review]
👍

**Pull request merged**